File: problems/70. Climbing Stairs/Solution.java
class Solution {
    public int climbStairs(int n) {
        int first = 1, second = 1, result = 1;
        for (int i = 2; i <= n; i++) {
            result = first + second;
            first = second;
            second = result;
        }
        return result;
    }
}

File: problems/70. Climbing Stairs/Solution.py
class Solution:
    def climbStairs(self, n: int) -> int:
        steps = [0] * 2
        steps[0] = 1
        steps[1] = 1
        for i in range(2, n + 1):
            current_steps = steps[0] + steps[1]
            steps[0] = steps[1]
            steps[1] = current_steps
        return steps[-1]